crates/sui/src/client_commands.rs

```diff
@@ -840,6 +851,47 @@ impl SuiClientCommands {
                 };
                 SuiClientCommandResult::Addresses(output)
             }
+            SuiClientCommands::Balance {
+                address,
+                with_coins,
+            } => {
+                let address = get_identity_address(address, context)?;
+                let client = context.get_client().await?;
+                let mut objects: Vec<Coin> = Vec::new();
+                let mut cursor = None;
+                loop {
+                    let response = client
+                        .coin_read_api()
+                        .get_all_coins(address, cursor, None)
+                        .await?;
+                    objects.extend(response.data);
+
+                    if response.has_next_page {
+                        cursor = response.next_cursor;
+                    } else {
+                        break;
+                    }
+                }
+                let mut coins_by_type: BTreeMap<String, (Option<SuiCoinMetadata>, Vec<Coin>)> =
```

Comment:
You should generally not need to annotate the map with its type like this -- once you have used it, Rust will infer what the types of keys and values are.

Reply:
Unfortunately, the compiler complains.
```
error[E0282]: type annotations needed for `BTreeMap<std::string::String, V>`
   --> crates/sui/src/client_commands.rs:881:21
    |
881 |                 let mut coins_by_type = BTreeMap::new();
    |                     ^^^^^^^^^^^^^^^^^
...
885 |                             e.get_mut().1.push(c);
    |                             ------------- type must be known at this point
    |
help: consider giving `coins_by_type` an explicit type, where the type for type parameter `V` is specified
    |
881 |                 let mut coins_by_type: BTreeMap<std::string::String, V> = BTreeMap::new();
    |                                      ++++++++++++++++++++++++++++++++++

error[E0609]: no field `1` on type `&mut _`
   --> crates/sui/src/client_commands.rs:885:41
    |
885 | ...                   e.get_mut().1.push(c);
    |                                   ^
```